source/extensions/tracers/opentelemetry/tracer.cc

```diff
@@ -174,6 +176,9 @@ Tracing::SpanPtr Tracer::startSpan(const Tracing::Config& config, const std::str
   new_span.setId(Hex::uint64ToHex(span_id));
   // Respect the previous span's sampled flag.
   new_span.setSampled(previous_span_context.sampled());
+  if (!previous_span_context.tracestate().empty()) {
+    new_span.setTracestate(std::string{previous_span_context.tracestate()});
```

Comment:
Is this tested?

Reply:
Yup, via `ParseSpanContextFromHeadersTest`. 

In `Driver::startSpan` in `opentelemetry_tracer_impl.cc`, if there is a propagation header present, it'll call this fn, `Tracer::startSpan` with the previous span context. Since we're manually adding the `traceparent` propagation header (and `tracestate`) in the  the test and then calling `driver_->startSpan`, the `tracestate` in the header is passed through in the `previous_span_context` here.